state/cockroachdb/cockroachdb_access.go

```diff
@@ -310,66 +405,68 @@ func (p *cockroachDBAccess) BulkDelete(ctx context.Context, req []state.DeleteRe
 	return err
 }
 
-func (p *cockroachDBAccess) ExecuteMulti(ctx context.Context, request *state.TransactionalStateRequest) error {
-	p.logger.Debug("Executing PostgreSQL transaction")
+func (c *CockroachDBAccess) ExecuteMulti(ctx context.Context, request *state.TransactionalStateRequest) error {
+	c.logger.Debug("Executing CockroachDB transaction")
 
-	tx, err := p.db.Begin()
+	tx, err := c.db.Begin()
 	if err != nil {
 		return err
 	}
 
-	for _, o := range request.Operations {
-		switch o.Operation {
-		case state.Upsert:
-			var setReq state.SetRequest
-
-			setReq, err = getSet(o)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-			err = p.Set(ctx, &setReq)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-		case state.Delete:
-			var delReq state.DeleteRequest
-
-			delReq, err = getDelete(o)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-			err = p.Delete(ctx, &delReq)
-			if err != nil {
-				tx.Rollback()
-				return err
+	performTx := func() error {
+		for _, o := range request.Operations {
+			switch o.Operation {
+			case state.Upsert:
+				var setReq state.SetRequest
+
+				setReq, err = getSet(o)
+				if err != nil {
+					return err
+				}
+
+				err = c.Set(ctx, &setReq)
```

Comment:
One more thing on this method, as well as all the Bulk* like BulkSet and BulkDelete... Not something added in this PR, but this is actually NOT using transactions, so it's not bulk and can't be rolled back in case of errors.

Although we do start a transaction (see `c.db.Begin()` above), the transaction is not passed to the methods that do Get, Set, Delete, etc. So it's not being used....

Reply:
Done.